FWCore/Framework/interface/EventSetupRecord.h

```diff
@@ -207,26 +205,32 @@ namespace edm {
     protected:
       template <template <typename> typename H, typename T, typename R>
       H<T> getHandleImpl(ESGetToken<T, R> const& iToken) const {
-        if
-          UNLIKELY(iToken.transitionID() != transitionID()) { throwWrongTransitionID(); }
-        assert(iToken.isInitialized());
+        if UNLIKELY (not iToken.isInitialized()) {
+          std::rethrow_exception(makeInvalidTokenException(this->key(), DataKey::makeTypeTag<T>()));
+        }
```

Comment:
@Dr15Jones I ended up re-using `makeInvalidTokenException()` because the exception message of that was about the same I had in mind. But that makes me ask ...